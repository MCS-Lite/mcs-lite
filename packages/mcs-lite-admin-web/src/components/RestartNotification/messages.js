import { defineMessages } from 'react-intl';

const messages = defineMessages({
  'RestartNotification.restartRequired': {
    id: 'RestartNotification.restartReqired',
    defaultMessage: '完成修改後，請重新啟動 MCS Lite 服務使設定生效。',
  },
  'RestartNotification.restart': {
    id: 'RestartNotification.restart',
    defaultMessage: '重新啟動',
  },
});

export default messages;
