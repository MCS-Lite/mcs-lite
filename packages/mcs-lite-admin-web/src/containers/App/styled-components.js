import styled from 'styled-components';

export const HeightContainer = styled.div`
  height: 100%;
`;

export const Fixed = styled.div`
  position: fixed;
  bottom: 20px;
  width: 100%;
`;

export const ToastContainer = styled.div`
  padding: 0 100px;

  > * {
    margin-top: 8px;
  }
`;
