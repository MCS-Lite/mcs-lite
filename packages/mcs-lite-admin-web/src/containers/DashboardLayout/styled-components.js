import styled from 'styled-components';
import P from 'mcs-lite-ui/lib/P';
import Logo from '../../components/Logo';

export const Container = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const Header = styled.header`
  flex-basis: 40px;
  flex-shrink: 0;
  box-sizing: border-box;
  background-color: ${props => props.theme.color.grayLight};
  border-bottom: 1px solid ${props => props.theme.base.bodyBackgroundColor};
  padding: 0 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const HeaderItemWrapper = styled.div`
  display: flex;
  height: 100%;

  > * {
    border-left: 1px solid ${props => props.theme.base.bodyBackgroundColor};

    &:last-child {
      border-right: 1px solid ${props => props.theme.base.bodyBackgroundColor};
    }
  }
`;

export const HeaderItem = styled.div`
  height: 100%;
  display: flex;
  padding: 0 15px;
  align-items: center;
  cursor: pointer;
  color: ${props => props.theme.color.primary};

  > ${P} {
    margin-left: 5px;
  }
`;

export const StyledLogo = styled(Logo)`
  height: 22px;
  padding: 9px 0;
`;

export const Body = styled.div`
  flex-grow: 1;
  display: flex;

  ${'' /* Remind: fixed for firefox: https://goo.gl/Rjqio3 */}
  min-height: 0;
`;

export const Nav = styled.nav`
  flex-basis: 150px;
  box-sizing: border-box;
  background-color: ${props => props.theme.color.grayLight};
  border-right: 1px solid ${props => props.theme.base.bodyBackgroundColor};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const NavItem = styled.div`
  width: : 100%;
  height: ${props => props.theme.height.normal};
  display: flex;
  padding: 0 20px;
  align-items: center;
  cursor: pointer;
`;

export const NavItemControl = styled(NavItem)`
  color: ${props => props.theme.color.primary};
  border-top: 1px solid ${props => props.theme.base.bodyBackgroundColor};

  > ${P} {
    margin-left: 5px;
  }
`;

export const Main = styled.main`
  flex-grow: 1;
  background-color: ${props => props.theme.color.white};
  overflow: auto;
`;
