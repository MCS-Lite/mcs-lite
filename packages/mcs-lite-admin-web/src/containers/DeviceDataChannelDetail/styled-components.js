import styled from 'styled-components';
import { MobileContentWrapper, Small, A } from 'mcs-lite-ui';

export const CardContainer = styled(MobileContentWrapper)`
  padding: 8px 8px 0 8px;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayBase};
`;

export const HistoryContainer = styled(MobileContentWrapper)`
  padding: 16px;
`;

export const ChartWrapper = styled.div`
  height: 300px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const HistoryHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 8px;
`;

export const ResetWrapper = styled(A)`
  display: flex;
  font-size: 24px;
  align-items: center;

  > *:first-child {
    margin-right: 4px;
  }
`;
