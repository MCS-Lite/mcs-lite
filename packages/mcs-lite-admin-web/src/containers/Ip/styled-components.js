import styled from 'styled-components';
import Button from 'mcs-lite-ui/lib/Button';

export const StyledButton = styled(Button)`
  margin-top: 10px;
`;

export const NaWrapper = styled.div`
  margin-top: 40px;
  text-align: center;
`;
