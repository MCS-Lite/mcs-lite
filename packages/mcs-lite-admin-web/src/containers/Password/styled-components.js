import styled from 'styled-components';
import { P, MobileContentWrapper } from 'mcs-lite-ui';

export const Container = styled(MobileContentWrapper)`
  padding: 16px 16px 56px 16px;

  > div + div {
    margin-top: 8px;
  }

  > div:last-child {
    margin-bottom: 16px;
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 4px;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  padding: 8px 16px;

  > * {
    flex: 1;
  }

  > *:first-child {
    margin-right: 8px;
  }
`;

export const StyledP = styled(P)`
  margin-top: 4px;
`;
