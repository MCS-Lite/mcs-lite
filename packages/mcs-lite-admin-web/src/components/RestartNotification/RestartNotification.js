import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Toast from 'mcs-lite-ui/lib/Toast';
import A from 'mcs-lite-ui/lib/A';
import { NAV_WIDTH } from '../../containers/DashboardLayout/styled-components';

const Container = styled.div`
  height: ${props => props.theme.height.normal};
  margin: -20px -20px 20px -20px;
`;

const Fixed = styled.div`
  position: fixed;
  right: 0;
  left: ${NAV_WIDTH}px;
`;

const StyledToast = styled(Toast)`
  border: none;
  box-shadow: none;

  > div {
    justify-content: space-between;
  }
`;

const RestartNotification = ({ onClick, getMessages: t }) =>
  <Container>
    <Fixed>
      <StyledToast kind="warning">
        <div>
          {t('restartRequired')}
        </div>
        <A onClick={onClick}>
          {t('restart')}
        </A>
      </StyledToast>
    </Fixed>
  </Container>;

RestartNotification.displayName = 'RestartNotification';
RestartNotification.propTypes = {
  // Props
  onClick: PropTypes.func.isRequired,

  // React-intl I18n
  getMessages: PropTypes.func.isRequired,
};

export default RestartNotification;
