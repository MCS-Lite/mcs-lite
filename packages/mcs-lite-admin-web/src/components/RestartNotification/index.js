import { withGetMessages } from 'react-intl-inject-hoc';
import RestartNotification from './RestartNotification';
import messages from './messages';

export default withGetMessages(messages, 'RestartNotification')(
  RestartNotification,
);
