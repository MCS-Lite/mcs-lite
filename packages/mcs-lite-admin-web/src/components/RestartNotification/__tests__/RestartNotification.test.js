import React from 'react';
import R from 'ramda';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import { theme } from 'mcs-lite-theme';
import RestartNotification from '../RestartNotification';

// jest.mock('mcs-lite-ui/lib/Toast');
jest.mock('mcs-lite-ui/lib/A');

it('should renders <RestartNotification> correctly', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <RestartNotification onClick={() => {}} getMessages={R.identity} />
    </ThemeProvider>,
  );

  expect(toJson(wrapper.find(RestartNotification))).toMatchSnapshot();
});
