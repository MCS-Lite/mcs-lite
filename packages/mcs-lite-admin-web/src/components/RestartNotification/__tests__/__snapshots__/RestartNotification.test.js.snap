// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <RestartNotification> correctly 1`] = `
.c4 {
  box-sizing: border-box;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2);
  border-radius: 3px;
  background-color: #FFFFFF;
  overflow: hidden;
}

.c7 {
  margin: 0;
  font-size: 1rem;
  color: currentColor;
}

.c3 {
  border: 1px solid #F39A1E;
  background-color: rgba(255,255,255,0.9);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 20px;
  color: #F39A1E;
  background-color: rgba(243,154,30,0.1);
}

.c0 {
  height: 32px;
  margin: -20px -20px 20px -20px;
}

.c1 {
  position: fixed;
  right: 0;
  left: 150px;
}

.c2 {
  border: none;
  box-shadow: none;
}

.c2 > .c5 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<RestartNotification
  getMessages={[Function]}
  onClick={[Function]}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Transition
            component={false}
            enter={
              Object {
                "opacity": 1,
                "translateY": 0,
              }
            }
            leave={
              Object {
                "opacity": 0.5,
                "translateY": -10,
              }
            }
            onEnter={[Function]}
            onLeave={[Function]}
            runOnMount={true}
          >
            <Component
              defaultStyles={
                Array [
                  Object {
                    "data": <Styled(Toast)
                      kind="warning"
                >
                      <div>
                            restartRequired
                      </div>
                      <A
                            color="primary"
                            onClick={[Function]}
                      >
                            restart
                      </A>
                </Styled(Toast)>,
                    "key": "StyledToast",
                    "style": Object {
                      "opacity": 0.5,
                      "translateY": -10,
                    },
                  },
                ]
              }
              didLeave={[Function]}
              styles={
                Array [
                  Object {
                    "data": <Styled(Toast)
                      kind="warning"
                >
                      <div>
                            restartRequired
                      </div>
                      <A
                            color="primary"
                            onClick={[Function]}
                      >
                            restart
                      </A>
                </Styled(Toast)>,
                    "key": "StyledToast",
                    "style": Object {
                      "opacity": Object {
                        "damping": 26,
                        "precision": 0.01,
                        "stiffness": 170,
                        "val": 1,
                      },
                      "translateY": Object {
                        "damping": 26,
                        "precision": 0.01,
                        "stiffness": 170,
                        "val": 0,
                      },
                    },
                  },
                ]
              }
              willEnter={[Function]}
              willLeave={[Function]}
            >
              <Styled(Toast)
                key="StyledToast"
                kind="warning"
                style={
                  Object {
                    "opacity": 0.5,
                    "undefined": "translateY(-10px) ",
                  }
                }
              >
                <Toast
                  className="c2"
                  kind="warning"
                  style={
                    Object {
                      "opacity": 0.5,
                      "undefined": "translateY(-10px) ",
                    }
                  }
                >
                  <Styled(Card)
                    className="c2"
                    kind="warning"
                    style={
                      Object {
                        "opacity": 0.5,
                        "undefined": "translateY(-10px) ",
                      }
                    }
                  >
                    <Card
                      className="c2 c3"
                      innerRef={undefined}
                      kind="warning"
                      style={
                        Object {
                          "opacity": 0.5,
                          "undefined": "translateY(-10px) ",
                        }
                      }
                    >
                      <div
                        className="c2 c3 c4"
                        kind="warning"
                        style={
                          Object {
                            "opacity": 0.5,
                            "undefined": "translateY(-10px) ",
                          }
                        }
                      >
                        <Styled(P)
                          kind="warning"
                        >
                          <P
                            className="c5 c6"
                            innerRef={undefined}
                            kind="warning"
                          >
                            <Component
                              className="c5 c6 c7"
                              kind="warning"
                            >
                              <div
                                className="c5 c6 c7"
                                kind="warning"
                              >
                                <div>
                                  restartRequired
                                </div>
                                <A
                                  color="primary"
                                  onClick={[Function]}
                                />
                              </div>
                            </Component>
                          </P>
                        </Styled(P)>
                      </div>
                    </Card>
                  </Styled(Card)>
                </Toast>
              </Styled(Toast)>
            </Component>
          </Transition>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RestartNotification>
`;
